### DISCUSSION TITLE
Support Cross-Origin-Opener-Policy and Cross-Origin-Embedder-Policy

### DISCUSSION THREAD

**Discussion by charlesneimog** [squidfunk/mkdocs-material#6056] [Q&A] [answered]
Hi, I am trying to use `SharedArrayBuffer` for some examples in `PdWebCompiler`. In others sites, like in ordinary Github Pages, this is possible using the script provide here: https://github.com/orgs/community/discussions/13309. But in Material Docs Page, there is an error where I have an infinite loop of reloading pages....

Someone have some info or some way to solve this? The _"bug"_ can be reproduced using the `shared-buffer` branch of https://github.com/charlesneimog/PdWebCompiler/tree/shared-buffer.

Then just run: `mkdocs server`.

Thanks for any help!

**Comment by alexvoss** [comment]
Hi, sorry for the silence but this question is a little more involved than most as it revolves around quite a complex integration of technologies. I may be holding the wrong end of the stick here but... 

a) Have you tried using `mkdocs serve --no-livereload` to turn off the reloading mechanism? Would that solve your problem?
b) Is the point of that code you point to not to insert a couple of response headers into the page when GitHub does not support supplying these? If it was possible to make `mkdocs serve` supply these headers, would that render Javascript  hack superfluous? If you need to host the result on Github then you would need to distinguish between the case where you use `mkdocs serve` and the version that gets built for GitHub.

**Comment by charlesneimog** [comment]
Hi @alexvoss, I didn't know what to do, now I have some things to research, thank you very much.

The `a` suggestion not work, but I will research a little more about `b` and post what I find here!

Thank you again!

**Reply by alexvoss** [reply]
If you are happy to patch MkDocs then you should be able to smuggle in the headers you need into the [LiveReloadServer class here](https://github.com/mkdocs/mkdocs/blob/00b648f0884244a139f034554b59a032113176f6/mkdocs/livereload/__init__.py#L243). Just adding to the array of headers passed should do the trick.

If you don't want to do that then perhaps a feature request to allow extra headers to be specified in the mkdocs.yml? I imagine there may be other use cases?

**Reply by squidfunk** [reply, marked as answer]
This should be taken upstream to MkDocs. The livereload capabilities are provided by [MkDocs](https://github.com/mkdocs/mkdocs), not by Material for MkDocs. If you wish to follow up on this, please [open a discussion](https://github.com/mkdocs/mkdocs/discussions) or an issue with a minimal reproduction on [MkDocs' issue tracker](https://github.com/mkdocs/mkdocs/issues).

**Reply by alexvoss** [reply]
Right, sorry, I should have said explicitly that I meant a feature request with MkDocs!
